cg/meta/workflow/prepare_fastq.py

```diff
@@ -0,0 +1,88 @@
+"""API to prepare fastq files for analysis"""
+
+import logging
+import os
+from typing import Dict
+from pathlib import Path
+
+from cg.meta.compress.compress import CompressAPI
+from cg.apps.crunchy import CrunchyAPI
+from cg.meta.compress import files
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.store import Store
+from housekeeper.store import models as hk_models
+
+LOG = logging.getLogger(__name__)
+
+
+class PrepareFastqAPI:
+    def __init__(self, store: Store, compress_api: CompressAPI):
+        self.store = store
+        self.hk_api: HousekeeperAPI = compress_api.hk_api
+        self.compress_api: CompressAPI = compress_api
+        self.crunchy_api: CrunchyAPI = compress_api.crunchy_api
+
+    def get_compression_objects(self, case_id: str) -> list:
+        """Return a list of compression objects"""
+        case_obj = self.store.family(case_id)
+        compression_objects = []
+        for link in case_obj.links:
+            sample_id = link.sample.internal_id
+            version_obj = self.compress_api.get_latest_version(sample_id)
+            compression_objects.extend(files.get_spring_paths(version_obj))
+        return compression_objects
+
+    def is_spring_decompression_needed(self, case_id: str) -> bool:
+        """Check if spring decompression needs to be started"""
+        compression_objects = self.get_compression_objects(case_id=case_id)
+        for compression_object in compression_objects:
+            if (
+                self.crunchy_api.is_compression_pending(compression_object)
+                or compression_object.pair_exists()
+            ):
+                continue
+            return True
+        return False
+
+    def is_spring_decompression_running(self, case_id: str) -> bool:
+        """Check if case is being decompressed"""
+        compression_objects = self.get_compression_objects(case_id=case_id)
+        for compression_object in compression_objects:
+            if self.crunchy_api.is_compression_pending(compression_object):
+                return True
+        return False
+
+    def start_spring_decompression(self, case_id: str, dry_run: bool) -> bool:
+        """Starts spring decompression"""
+        compression_objects = self.get_compression_objects(case_id=case_id)
```

Comment:
More type hints! Makes the code som much more easy to understand
```suggestion
        compression_objects: List[CompressionData] = self.get_compression_objects(case_id=case_id)
```